tests/tekton-tasks_test.go

```diff
@@ -33,54 +35,106 @@ var _ = Describe("Tekton-tasks", func() {
 
 		It("[test_id:TODO]operator should not create any cluster tasks", func() {
 			liveTasks := &pipeline.ClusterTaskList{}
-			Eventually(func() bool {
-				err := apiClient.List(ctx, liveTasks,
-					client.MatchingLabels{
-						tektontasks.TektonTasksVersionLabel: operands.TektonTasksVersion,
-					},
-				)
-				Expect(err).ToNot(HaveOccurred())
-				return len(liveTasks.Items) == 0
-			}, tenSecondTimeout, time.Second).Should(BeTrue(), "tasks should not exists")
+
+			err := apiClient.List(ctx, liveTasks,
+				client.MatchingLabels{
+					common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			//Nothing to check, end the test successfully
+			if len(liveTasks.Items) == 0 {
+				return
+			}
+
+			err = apiClient.Delete(ctx, &liveTasks.Items[0])
+			Expect(err).ToNot(HaveOccurred())
+
+			deletedTask := &pipeline.ClusterTask{}
+			Consistently(func() metav1.StatusReason {
+				err := apiClient.Get(ctx, client.ObjectKeyFromObject(&liveTasks.Items[0]), deletedTask)
+				Expect(err).To(HaveOccurred())
+				return errors.ReasonForError(err)
```

Comment:
IMO I prefer `errors.IsNotFound(err)` but I leave it up to you!

Reply:
In this case, it might be better to use `ReasonForError` because when the error is different, it will show you what typer of error it is (better for debugging). In `errors.IsNotFound(err)` it will show you just true / false.